Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -1045,18 +1046,37 @@ internal ItemImageInfo[] DeserializeImages(string value)
                 return Array.Empty<ItemImageInfo>();
             }
 
-            var list = new List<ItemImageInfo>();
-            foreach (var part in value.SpanSplit('|'))
+            // TODO The following is an ugly performance optimization, but it's extremely unlikely that the data in the database would be malformed
+            var valueSpan = value.AsSpan();
+            var count = valueSpan.CountOccurrences('|') + 1;
```

Comment:
I was debating it myself. I removed it and re-added it. It saves a lot of list allocations because it's extremely unlikely that any of the strings are invalid. I couldn't make it fail without altering the db manually. This method is a very hot path.